Preconditioner factory does not compile with DUNE master

To replicate build the tests:
```

/home/mblatt/src/dune/opm-release-master/opm-simulators/opm/simulators/linalg/PreconditionerFactory.hpp: In lambda function:
/home/mblatt/src/dune/opm-release-master/opm-simulators/opm/simulators/linalg/PreconditionerFactory.hpp:321:34: error: ‘SeqILU0’ does not name a type
                 using Smoother = SeqILU0<M, V, V>;
                                  ^~~~~~~
/home/mblatt/src/dune/opm-release-master/opm-simulators/opm/simulators/linalg/PreconditionerFactory.hpp:322:46: error: ‘Smoother’ was not declared in this scope
                 return makeAmgPreconditioner<Smoother>(op, prm, true);
                                              ^~~~~~~~
/home/mblatt/src/dune/opm-release-master/opm-simulators/opm/simulators/linalg/PreconditionerFactory.hpp:322:46: note: suggested alternative: ‘smoother’
                 return makeAmgPreconditioner<Smoother>(op, prm, true);
                                              ^~~~~~~~
                                              smoother
/home/mblatt/src/dune/opm-release-master/opm-simulators/opm/simulators/linalg/PreconditionerFactory.hpp:336:34: error: ‘SeqILUn’ does not name a type
                 using Smoother = SeqILUn<M, V, V>;
                                  ^~~~~~~
/home/mblatt/src/dune/opm-release-master/opm-simulators/opm/simulators/linalg/PreconditionerFactory.hpp:337:46: error: ‘Smoother’ was not declared in this scope
                 return makeAmgPreconditioner<Smoother>(op, prm, true);
```

This bug is also part of the release.

--- a/opm/simulators/linalg/PreconditionerFactory.hpp
+++ b/opm/simulators/linalg/PreconditionerFactory.hpp
@@ -318,7 +318,11 @@ private:
 	doAddCreator("kamg", [](const O& op, const P& prm, const std::function<Vector()>&) {
             const std::string smoother = prm.get<std::string>("smoother", "ParOverILU0");
             if (smoother == "ILU0" || smoother == "ParOverILU0") {
+#if DUNE_VERSION_NEWER(DUNE_ISTL, 2, 7)
+                using Smoother = SeqILU<M, V, V>;
+#else
                 using Smoother = SeqILU0<M, V, V>;
+#endif
                 return makeAmgPreconditioner<Smoother>(op, prm, true);
             } else if (smoother == "Jac") {
                 using Smoother = SeqJac<M, V, V>;
@@ -333,7 +337,11 @@ private:
                 using Smoother = SeqSSOR<M, V, V>;
                 return makeAmgPreconditioner<Smoother>(op, prm, true);
             } else if (smoother == "ILUn") {
+#if DUNE_VERSION_NEWER(DUNE_ISTL, 2, 7)
+                using Smoother = SeqILU<M, V, V>;
+#else
                 using Smoother = SeqILUn<M, V, V>;
+#endif
                 return makeAmgPreconditioner<Smoother>(op, prm, true);
             } else {
                 OPM_THROW(std::invalid_argument, "Properties: No smoother with name " << smoother <<".");

